File: hailhaydra/src/main/java/com/yeqiu/hailhaydra/activity/BaseActivity.java
package com.yeqiu.hailhaydra.activity;

import android.view.View;

import com.yeqiu.hailhaydra.R;
import com.yeqiu.hydra.basecontroller.HydraBaseActivity;

/**
 * @project：HailHydra
 * @author：小卷子
 * @date 2018/9/15
 * @describe：
 * @fix：
 */
public abstract class BaseActivity extends HydraBaseActivity {


    @Override
    protected int getDefHeadBackImgId() {

        return R.drawable.icon_head_back;
    }

    @Override
    protected void initImmersionBar() {
        super.initImmersionBar();
    }


    @Override
    public void onStatusClick(View view) {

    }



}

File: sample/src/main/java/com/fastmvvm/sample/view/activity/BaseActivity.kt
package com.fastmvvm.sample.view.activity

import android.widget.TextView
import androidx.databinding.ViewDataBinding
import com.afollestad.materialdialogs.MaterialDialog
import com.afollestad.materialdialogs.customview.customView
import com.afollestad.materialdialogs.customview.getCustomView
import com.afollestad.materialdialogs.lifecycle.lifecycleOwner
import com.fastmvvm.sample.R
import com.yeqiu.fastmvvm.view.ViewModelDataBindingActivity
import com.yeqiu.fastmvvm.viewmodel.BaseViewModel

/**
 * @project：FastMvvm
 * @author：小卷子
 * @date 2023/7/9
 * @describe：
 * @fix：
 */
abstract class BaseActivity<VM : BaseViewModel, VB : ViewDataBinding> : ViewModelDataBindingActivity<VM,VB>() {


    private lateinit var loadingDialog: MaterialDialog


    override fun showLoading(message:String) {


        if (!::loadingDialog.isInitialized) {
            loadingDialog = MaterialDialog(this)
                .cancelable(true)
                .cancelOnTouchOutside(false)
                .cornerRadius(12f)
                .customView(R.layout.dialog_loading)
                .lifecycleOwner(this)
            loadingDialog.getCustomView().run {
                (findViewById<TextView>(R.id.loading_tips)).text =message
            }
        }

        loadingDialog.show()
    }

    override fun dismissLoading() {

        if (::loadingDialog.isInitialized){
            loadingDialog.dismiss()
        }

    }
}